src/core/api/public_api.ts

```diff
@@ -731,11 +731,19 @@ class Player extends EventEmitter<IPublicAPIEvent> {
     // inilialize `_priv_playing$` to false (meaning the content is not playing yet)
     this._priv_playing$.next(false);
 
-    const videoElement = this.videoElement;
 
+    const videoElement = this.videoElement;
+    const _priv_currentTimeHandler = {
+      isSeekingFromInside: false,
+      getCurrentTime: () => videoElement.currentTime,
+      setCurrentTime(time: number) {
+        this.isSeekingFromInside = true;
+        videoElement.currentTime = time
+      }
```

Comment:
Why not just declaring all that in the `clock$`?

Why attaching to the private state of the public API? It doesn't seem to be needed in any other function

Reply:
I was doing that in the first version but we saw in a meeting that it would be nicer to not put both methods in the clock tick.

Then, I need that method (`setCurrentTime`) in both `initializeMediaSourcePlayback` and `initDirectFile`

Am I still right?

Reply:
Yes my bad.
I was speeking about the `clock$` when what I really meant was in `src/core/api/clock.ts`, not in the `clock$` itself.

Reply:
>Then, I need that method (setCurrentTime) in both initializeMediaSourcePlayback and initDirectFile

Yes I understand. But they are not part of the `RxPlayer` class defined in `public_api.ts` which is why I asked why would we attach it to its private state.

Reply:
Done